module/zfs/dmu_object.c

```diff
@@ -32,6 +32,15 @@
 #include <sys/zfeature.h>
 #include <sys/dsl_dataset.h>
 
+/*
+ * Each of the concurrent object allocators will grab
+ * 2^dmu_object_alloc_chunk_shift objects at a time.  The default is to
+ * grab 128 object, which s 4 blocks worth.  This was experimentally
+ * determined to be the lowest value that eliminates the measurable effect
+ * of lock contention from this code path.
+ */
+int dmu_object_alloc_chunk_shift = 7;
```

Comment:
"dnode slots" or "object slots" would be clearer to me than "objects".  I also just noticed a typo in this comment,  `s/which s 4/which is 4/`.

"The default is to grab 128 object slots, which is 4 blocks worth."